src/wallpaperController.ts

```diff
@@ -450,32 +450,25 @@ class WallpaperController {
     /**
      * Set an existing history entry as wallpaper.
      *
-     * @param {string} historyId Unique ID
+     * @param {HistoryModule.HistoryEntry} historyEntry The history entry to set as wallpaper.
      */
-    async setWallpaper(historyId: string): Promise<void> {
-        const historyElement = this._historyController.get(historyId);
-
-        if (historyElement?.id && historyElement.path && this._historyController.promoteToActive(historyElement.id)) {
```

Comment:
@Lucki do you know if there was a particular reason for having the `promotToActive` call here in the condition? To me it looked superfluous.

Reply:
What do you mean? It pushes the selected wallpaper back to the top of the history. This ensures that the current active wallpaper is on the (old) "current" menu element and that a popular wallpaper the user switched to doesn't get removed by being at the end of the list.

Reply:
Oh, I see it's down there again. Maybe an oversight… both have changes in a big pre-TS commit.